index.js

```diff
@@ -1,95 +1,129 @@
 // ***** Question 1 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
+let counter = 1
+counter += 1
 
-// console.log(counter) 
-// // => 2
-// console.log("%c----------", "color: red") 
+console.log("%cQuestion 1", "color: red") 
+
+console.log(counter) 
+
+console.log("%c----------", "color: red") 
 
 
 // ***** Question 2 *****
 
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
 
-// console.log(name)
-// // => "Ian"
+console.log("%cQuestion 2", "color: red") 
+const name = "Demetrio"
+console.log(name)
+// => "Ian"
 // name = "Not Ian" 
-// // => TypeError
-// console.log("%c----------", "color: red") 
+// => TypeError
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red") 
 
-// console.log(drinkWater(12))
-// // => "Man I sure am thirsty"
-// // => "Ahh that hits the spot"
-// // => 11
-// console.log("%c----------", "color: red") 
+console.log(drinkWater(12))
+
+function drinkWater(current_thirst_level){
+  console.log("Man I sure am thirsty")
+  current_thirst_level -= 1
+  console.log("Ahh that hits the spot")
+  console.log(current_thirst_level)
+}
+// => "Man I sure am thirsty"
+// => "Ahh that hits the spot"
+// => 11
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red") 
+
+console.log(sameSameButDifferent(5, 5))
+
+function sameSameButDifferent(num, maybeNum){
+  if(num == maybeNum) {
+  console.log("same same")
+  } if ( num === maybeNum){
+    console.log("same same (but different)")
+  } else {
+    console.log("different")
+  } }
```

Comment:
This function isn't quite working correctly. The `==` should be used for the "same same (but different)" condition, since that will compare a number and a string by converting the string to a number. 

Also, same comment as above - this function needs to **return** a value, not just `console.log` something.

Lastly, make sure your indentation is consistent - it's good to develop the habit of indenting your code early so it becomes second nature!

```js
function sameSameButDifferent(num, maybeNum){
  if(num === maybeNum) {
    return "same same"
  } if ( num == maybeNum){
    return "same same (but different)"
  } else {
    return "different"
  } 
}
```